pkg/virt-controller/network/pod_annotations_test.go

```diff
@@ -136,9 +152,16 @@ var _ = Describe("pod annotations", func() {
 				libvmi.InterfaceDeviceWithSRIOVBinding("doo"),
 			}
 			podAnnotationMap := network.GeneratePodAnnotations(networks, interfaces, networkStatus, bindingPlugins)
-			Expect(podAnnotationMap).To(Equal(map[string]string{
-				downwardapi.NetworkInfoAnnot:  `{"interfaces":[{"network":"foo","deviceInfo":{"type":"pci","version":"1.0.0","pci":{"pci-address":"0000:65:00.2"}}}]}`,
-				deviceinfo.NetworkPCIMapAnnot: `{"doo":"0000:65:00.3"}`,
+			Expect(podAnnotationMap).To(HaveLen(1))
```

Comment:
Why is this needed?

Reply:
We cannot use Equal because the annotation value consist of JSON array that is order could change, JSONMatch matcher doesnt help neither.

I added this len check to make sure no other annotation were added beside the network-info one this test  expects. I think it make it closer to the Equal matcher.

Reply:
The HaveKey doesnt ensure there is one an only key, by asserting HaveLen(1) we make sure only the expected annotation exist.

Reply:
Ok, but I do not like this one. I would have preferred a normalization to be done over this.
But it is not worth the effort.